bundles/org.openhab.binding.xmppclient/src/main/java/org/openhab/binding/xmppclient/handler/XMPPClientHandler.java

```diff
@@ -0,0 +1,116 @@
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+import org.openhab.binding.xmppclient.action.XMPPActions;
+import org.openhab.binding.xmppclient.internal.XMPPClient;
+import org.openhab.binding.xmppclient.handler.XMPPClientConfiguration;
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseBridgeHandler;
+import org.eclipse.smarthome.core.thing.binding.ThingHandlerService;
+import org.eclipse.smarthome.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+
+/**
+ * The {@link XMPPClientHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Pavel Gololobov - Initial contribution
+ */
+
+public class XMPPClientHandler extends BaseBridgeHandler {
+    private final Logger logger = LoggerFactory.getLogger(XMPPClientHandler.class);
+    private XMPPClient xmppClient;
+    private XMPPClientConfiguration config;
+    private final Map<ChannelUID, PublishTriggerChannel> channelStateByChannelUID = new HashMap<>();
+
+    public XMPPClientHandler(Bridge thing) {
+        super(thing);
+    }
+
+    public XMPPClient getXMPPClient() {
+        return xmppClient;
+    }
+
+    @Override
+    public Collection<Class<? extends ThingHandlerService>> getServices() {
+        return Collections.singleton(XMPPActions.class);
+    }
+
+    @Override
+    protected void triggerChannel(ChannelUID channelUID, String event) {
+        super.triggerChannel(channelUID, event);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        // not supported
+    }
+
+    @Override
+    public void initialize() {
+        updateStatus(ThingStatus.UNKNOWN);
+        scheduler.schedule(this::doConnect, 0, TimeUnit.SECONDS);
+    }
+
+    @Override
+    public void dispose() {
+        channelStateByChannelUID.values().forEach(c -> c.stop());
+        channelStateByChannelUID.clear();
+        xmppClient.disconnect();
+        super.dispose();
+    }
+
+    private void doConnect() {
+        config = getConfigAs(XMPPClientConfiguration.class);
+        xmppClient = new XMPPClient();
+
+        String resource = "openHAB";
+        if((config.resource != null) && !config.resource.isEmpty()) {
+            resource = config.resource;
+        }
+
+        try {
+            xmppClient.connect(config.host, config.port, config.username, config.domain, config.password, resource);
+        } catch (SmackException | IOException | XMPPException e) {
+            logger.info("XMPP connection error", e);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, e.getMessage());
```

Comment:
Is this always a CONFIGURATIOB_ERROR? maybe COMMUNICATION_ERROR would be a better fit.